tapir/shifts/views/views.py

```diff
@@ -309,3 +314,210 @@ def get(self, request, *args, **kwargs):
         call_command("run_freeze_checks")
         messages.info(request, _("Frozen statuses updated."))
         return super().get(request, args, kwargs)
+
+
+class SolidarityShiftUsed(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        date = datetime.now()
+        tapir_user = get_object_or_404(TapirUser, pk=kwargs["pk"])
```

Comment:
We should check that either the user using the solidarity shift is the same as the one doing the request (`request.user.pk == tapir_user.pk`), or that the user doing the request has the relevant rights (`request.user.has_perm(PERMISSION_SHIFTS_MANAGE)`)